docs/cep/windows/window-types/tumbling-external-time.md

```diff
@@ -0,0 +1,56 @@
+---
+title: WINDOW TUMBLING_EXTERNAL_TIME()
+---
+
+A batch (tumbling) time window based on external time, that holds events arrived during `windowTime` periods, and gets updated for every `windowTime`.
+
+## Syntax
+
+    WINDOW TUMBLING_EXTERNAL_TIME(<LONG> timestamp, <INT|LONG|TIME> time)
+    WINDOW TUMBLING_EXTERNAL_TIME(<LONG> timestamp, <INT|LONG|TIME> time, <INT|LONG|TIME> start)
+    WINDOW TUMBLING_EXTERNAL_TIME(<LONG> timestamp, <INT|LONG|TIME> time, <INT|LONG|TIME> start, <INT|LONG|TIME> timeout)
+    WINDOW TUMBLING_EXTERNAL_TIME(<LONG> timestamp, <INT|LONG|TIME> time, <INT|LONG|TIME> start, <INT|LONG|TIME> timeout, <BOOL> replace.with.batchtime)
+
+## Query Parameters
+
+| Name                   | Description        | Default Value           | Possible Data Types | Optional | Dynamic |
+|--------------|---------------------------------------|--------------------------|------------------|----------|---------|
+| timestamp              | The time which the window determines as current time and will act upon. The value of this parameter should be monotonically increasing. |                        | LONG                | No       | Yes     |
+| time            | The batch time period for which the window should hold events.    |                      | INT LONG TIME       | No       | No      |
+| start             | User defined start time. This could either be a constant (of type `int`, `long`, or `time`) or an attribute of the corresponding stream (of type `long`). If an attribute is provided, initial value of attribute would be considered as startTime. | Timestamp of first event     | INT LONG TIME       | Yes      | Yes     |
+| timeout      | Time to wait for arrival of new event, before flushing and giving output for events belonging to a specific batch.    | System waits till an event from next batch arrives to flush current batch | INT LONG TIME       | Yes      | No      |
+| replace.with.batchtime | This indicates to replace the expired event timeStamp as the batch end timeStamp        | System waits till an event from next batch arrives to flush current batch | BOOL          | Yes      | No      |
+
+## Example 1
+
+```sql
+CREATE STREAM cseEventStream (symbol string, price float, volume int);
```

Comment:
@StoyanTc and @koshy-thomas, this stream worker is not valid. The other two examples on this page probably have the same eventTime issue. Ideally, users would be able to have a stream worker that they could try all the window examples in this page on. 

Either that, or we need to give very clear instructions, and maybe examples, of what to put in eventTime.

Reply:
@diana-macrometa you can refer to the previous comment of eventTime.. we can take it from cseEventStream itself.. and add a note that it needs to be unix timestamp in millis.. Let me know if you have doubts on that.. And I still think that outputStream definition is needed most likey but once Stoyan gets some time he can check and confirm I guess

Reply:
@diana-macrometa - The code below should work:
```
-- Metadata with name, description, and query language version.
@App:name("Temperature-Analytics")
@App:description("This stream worker creates a stream and query.")
@App:qlVersion("2")

CREATE STREAM cseEventStream (symbol string, price float, volume int, eventTime long);
CREATE SINK outputStream WITH (type='stream', stream='outputStream') (symbol string, price float);
CREATE WINDOW cseEventWindow (symbol string, price float, volume int, eventTime long) SLIDING_EXTERNAL_TIME(eventTime, 20 sec) OUTPUT expired events;

@info(name = 'query0')
INSERT INTO cseEventWindow
SELECT symbol, price, volume, eventTimestamp() AS eventTime
FROM cseEventStream;

@info(name = 'query1')
INSERT expired events INTO outputStream
SELECT symbol, price
FROM cseEventWindow;
```

Reply:
Thanks, @StoyanTc! @koshy-thomas paired with me and helped me figure out what I was doing wrong.